src/Dropdown.elm

```diff
@@ -358,111 +358,83 @@ withListAttributes attrs (Config config) =
-                                |> modifySelectedItems config.dropdownType item
-                    in
-                    ( { state
-                        | isOpen = closeOnlyIfNotMultiSelect config state
-                      }
-                    , updateSelectedItemsCommand config.onSelectMsg selectedItemsNew
-                    )
-
-                OnFilterTyped val ->
-                    ( { state | filterText = val }, Cmd.none )
-
-                OnKeyDown key ->
-                    let
-                        newIndex =
-                            case key of
-                                ArrowUp ->
-                                    if state.focusedIndex > 0 then
-                                        state.focusedIndex - 1
-
-                                    else
-                                        0
-
-                                ArrowDown ->
-                                    if state.focusedIndex < List.length items - 1 then
-                                        state.focusedIndex + 1
-
-                                    else
-                                        List.length items - 1
-
-                                _ ->
-                                    state.focusedIndex
-
-                        isOpen =
-                            case key of
-                                Esc ->
-                                    False
-
-                                Enter ->
-                                    False
-
-                                _ ->
-                                    True
-
-                        maybeFocusedItem =
-                            items
-                                |> List.indexedMap (\i item -> ( i, item ))
-                                |> List.filter (\( i, _ ) -> i == state.focusedIndex)
-                                |> List.head
-                                |> Maybe.map Tuple.second
-
-                        cmd =
-                            case key of
-                                Enter ->
-                                    case maybeFocusedItem of
-                                        Just focusedItem ->
-                                            selectedItems
-                                                |> modifySelectedItems config.dropdownType focusedItem
-                                                |> updateSelectedItemsCommand config.onSelectMsg
-
-                                        Nothing ->
-                                            Cmd.none
-
-                                _ ->
-                                    Cmd.none
-                    in
-                    ( { state | focusedIndex = newIndex, isOpen = isOpen }, cmd )
+        clampIndex i =
+            clamp 0 (List.length items - 1) i
     in
-    ( State newState, newCommand )
+    case key of
+        ArrowDown ->
+            ( State { state | focusedIndex = clampIndex <| state.focusedIndex - 1 }, Cmd.none )
```

Comment:
I think you reversed the operation here and in the `ArrowUp`, so the `ArrowDown` should move the cursor forward in the list

Reply:
Also, we want to open (`isOpen = True`) when `ArrowDown` or `ArrowUp`, otherwise we break navigation without the mouse. The same for `Enter`:
```elm
            if state.isOpen then
                -- What yours does...
                
            else
                ( State { state | isOpen = True }, Cmd.none )
```

Reply:
I switched the operations, indeed. Will fix it.

Reply:
I am confused, @PedroHLC . What is the behavior we want?

Reply:
Do I understand correctly that, if you press ArrowUp, ArrowDown, or Enter when the state.isOpen == False, you just want to set the state.isOpen to True?

Reply:
I will let the code talk for itself. There you can see how I understood you want it to work.